lib/Ix/DBIC/ResultSet.pm

```diff
@@ -471,6 +449,29 @@ sub _ix_check_user_properties (
       }
     }
 
+    if ($date_fields{$prop}) {
+      unless (defined $value) {
+        if ($prop_info->{$prop}->{is_optional}) {
+          $user_prop{$prop} = $value;
+          next PROP;
+        } else {
+          $property_error{$prop} = "no value given for required field";
+          next PROP;
+        }
+      }
+
+      # Already a DateTime object (checked above)?
+      unless (ref $value) {
+        if (my $dt = parsedate($value)) {
+          # great, it's already valid
+          $value = $dt;
+        } else {
+          $property_error{$prop} = "invalid date value";
+          next PROP;
+        }
+      }
+    }
+
```

Comment:
Perhaps the above block should be reordered so any valid cases fall through for other tests that may be added (instead of next-ing out on line 456)?